src/site/markdown/index.md

```diff
@@ -340,8 +386,53 @@ If the pixels are ints, that’s  an 8 GB file.  We can do
 ```
     
 The `reset()` method causes the internal stream to seek to the beginning of the data area.
-If that’s not possible it returns false. 
+If that’s not possible it returns false.
+
+#### Random Access
+It is possible to specify alternate random access by implementing the `RandomAccessFileIO` interface.  This is useful
+when a Fits file isn't necessarily on disk, but rather in object storage on the web, for example, or any other
+network access.
 
+```java
+import nom.tam.util.RandomAccessFileIO;
+
+import java.io.FileDescriptor;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+
+public final class S3RandomAccessFileIO implements RandomAccessFileIO {
```

Comment:
Perhaps just keep this line in your example, and replace the body with just ellipses (`...`). Also, currently you reference `S3RandomAccessFileIO` in the README earlier, so it's too late to specify it here what it is. Either define it earlier (preferably just before the examples that refer to it), or use a more vanilla RandomAccess implementation in those examples....